jsonschema/tests/test_validators.py

```diff
@@ -775,11 +775,11 @@ def test_it_resolves_local_refs(self):
             self.assertEqual(resolved, self.referrer["properties"]["foo"])
 
     def test_it_resolves_local_refs_with_id(self):
-        schema = {"id": "foo://bar/schema#", "a": {"foo": "bar"}}
+        schema = {"id": "http://bar/schema#", "a": {"foo": "bar"}}
         resolver = RefResolver.from_schema(schema)
         with resolver.resolving("#/a") as resolved:
             self.assertEqual(resolved, schema["a"])
-        with resolver.resolving("foo://bar/schema#/a") as resolved:
+        with resolver.resolving("http://bar/schema#/a") as resolved:
```

Comment:
I had to change these to `http` because (at least on py3) the behaviour of `urljoin` changes based on the scheme.  With an unknown scheme it was not replacing the fragment at all.